pkg/provisioner/ironic/testserver/redfishvirtualmedia.go

```diff
@@ -0,0 +1,33 @@
+package testserver
+
+import (
+	"net/http"
+	"testing"
+)
+
+// RedfishVirtualMediaMock is a test server that implements Redfish Virtual Media semantics
+type RedfishVirtualMediaMock struct {
+	*MockServer
+}
+
+// NewRedfishVirtualMedia builds a redfish virtual media mock server
+func NewRedfishVirtualMedia(t *testing.T) *RedfishVirtualMediaMock {
+	return &RedfishVirtualMediaMock{
+		MockServer: NewTLS(t, "redfish-virtual-media"),
+	}
+}
+
+// Endpoint returns the URL to the server
+func (m *RedfishVirtualMediaMock) Endpoint() string {
+	if m == nil {
+		return "https://bmcaddress"
+	}
+	return m.MockServer.RawEndpoint()
+
+}
+
+// Ready configures the server with a valid response for /redfish/v1
+func (m *RedfishVirtualMediaMock) Ready() *RedfishVirtualMediaMock {
+	m.ResponseWithCode("/redfish/v1", "{}", http.StatusOK)
```

Comment:
The payload must be properly filled to let gofish unmarshal correctly the server answer. I've tried this one https://redfish.dmtf.org/redfish/mockups/v1/1018 and it works ok.

Reply:
Note: no need to specify the auth, default answer of MockServer will take care of it (otherwise you'll need to mock an additional call).

Reply:
Sure I will update the payload.

Reply:
@andfasano I've done the changes you requested and updated the `Ready` to
```
// Ready configures the server with a valid response for /redfish/v1
func (m *RedfishVirtualMediaMock) Ready() *RedfishVirtualMediaMock {
	m.ResponseWithCode("redfish/v1", `
	{
		"@odata.type": "#ServiceRoot.v1_9_0.ServiceRoot",
		"Id": "RootService",
		"Name": "Root Service",
		"RedfishVersion": "1.6.0",
		"UUID": "92384634-2938-2342-8820-489239905423",
		"Systems": {
			"@odata.id": "/redfish/v1/Systems"
		},
		"Chassis": {
			"@odata.id": "/redfish/v1/Chassis"
		},
		"Managers": {
			"@odata.id": "/redfish/v1/Managers"
		},
		"Tasks": {
			"@odata.id": "/redfish/v1/TaskService"
		},
		"SessionService": {
			"@odata.id": "/redfish/v1/SessionService"
		},
		"AccountService": {
			"@odata.id": "/redfish/v1/AccountService"
		},
		"EventService": {
			"@odata.id": "/redfish/v1/EventService"
		},
		"Registries": {
			"@odata.id": "/redfish/v1/Registries"
		},
		"UpdateService": {
			"@odata.id": "/redfish/v1/UpdateService"
		},
		"CertificateService": {
			"@odata.id": "/redfish/v1/CertificateService"
		},
		"Links": {
			"Sessions": {
				"@odata.id": "/redfish/v1/SessionService/Sessions"
			}
		},
		"Oem": { },
		"@odata.id": "redfish/v1/"
	}`, http.StatusOK)
	return m
}
```
I'm still getting 
```
=== RUN   TestValidateVirtualMedia
    server.go:31: redfish-virtual-media: new server created
    server.go:160: redfish-virtual-media: adding response for [GET] redfish/v1/
    virtualmedia_validation_test.go:18: could not connect to redfish BMC
--- FAIL: TestValidateVirtualMedia (1.12s)
```

Reply:
@iurygregory not sure if it's my problem but I still see the old code in the `validate_bootinterface` branch, can you please check it? 
(On my local machine, with the currently discussed changes, the test reaches the point where the call to `/redfish/v1/Systems` needs to be mocked to move on, but the connection and auth was ok)

Reply:
Thanks for the help @andfasano ! I've added some test scenarios